admin/spam/views.py

```diff
@@ -1,47 +1,141 @@
 from django.shortcuts import render
-from django.core.paginator import Paginator, EmptyPage, PageNotAnInteger
 from django.contrib.auth.decorators import login_required
+from django.views.generic import FormView, ListView
+from django.utils.decorators import method_decorator
+from django.core.urlresolvers import reverse
+from django.views.defaults import page_not_found
 
 from modularodm import Q
 from website.project.model import Comment
 
 from .serializers import serialize_comment
+from .forms import ConfirmForm
 
 
-def get_spam_list():
-    query = (
-        Q('reports', 'ne', {}) &
-        Q('reports', 'ne', None)
-    )
-    return Comment.find(query).sort('date_created')
+class SpamList(ListView):
+    template_name = 'spam/spam_list.html'
+    paginate_by = 10
+    paginate_orphans = 1
+    ordering = 'date_created'
 
+    def __init__(self):
+        self.status = str(Comment.FLAGGED)
+        super(SpamList, self).__init__()
 
-@login_required
-def spam_list(request):
-    paginator = Paginator(get_spam_list(), 10)
-
-    page_number = request.GET.get('page', 1)
-    try:
-        page = paginator.page(page_number)
-    except PageNotAnInteger:
-        page = paginator.page(1)
-    except EmptyPage:
-        page = paginator.page(paginator.num_pages)
-    context = {
-        'spam': map(serialize_comment, page),
-        'page': page,
-        'page_number': page_number,
-    }
-    return render(request, 'spam/spam.html', context)
+    def get_queryset(self):
+        self.status = self.request.GET.get('status', u'1')
+        query = (
+            Q('reports', 'ne', {}) &
+            Q('reports', 'ne', None) &
+            Q('spam_status', 'eq', int(self.status))
+        )
+        return Comment.find(query).sort(self.ordering)
 
+    def get_context_data(self, **kwargs):
+        queryset = kwargs.pop('object_list', self.object_list)
+        page_size = self.get_paginate_by(queryset)
+        paginator, page, queryset, is_paginated = self.paginate_queryset(
+            queryset, page_size)
+        return {
+            'spam': map(serialize_comment, queryset),
+            'page': page,
+            'status': self.status,
+            'page_number': page.number
+        }
 
-@login_required
-def spam_detail(request, spam_id):
-    context = {
-        'comment': serialize_comment(Comment.load(spam_id)),
-        'page_number': request.GET.get('page', 1),
-    }
-    return render(request, 'spam/comment.html', context)
+
+class UserSpamList(SpamList):
+    template_name = 'spam/user.html'
+
+    def __init__(self):
+        self.user_id = None
```

Comment:
Why is it necessary to set `user_id` as an attribute; you can always access it from `self.kwargs`.